test/test_autograd.py

```diff
@@ -3333,6 +3333,343 @@ def test_checkpointing(self):
         mean_combined = torch.stack(feat_combined).mean()
         mean_combined.backward()
 
+    def test_autograd_views_codegen(self):
+        # This is not necessarily the absolute correct behavior, but this is the current
+        # one. This test is here to make sure that any change to this behavior is detected
+        # and not silent. The TODOs below mark the places with unexpected behavior.
+        # Note that any change in these test will be BC-breaking and should be done carefully.
+
+        # This test checks the behavior of two codegen functions (view_as and unbind)
+        # with respect to view tracking and inplace operation on the output.
+
+        def run_test(grad_mode, requires_grad, is_view, should_raise_tuple):
```

Comment:
The ordering of the arguments here is a little backward. First you allocate the variable (using `requires_grad`), then you run the operation with grad enabled or disabled (`grad_mode`). But in the argument list here they're reverse order

Reply:
In every usage of this function, `is_view` is true. I don't think we're ever going to make `view_as` or `unbind` return not views? So why have them be arguments to this function?

Reply:
Well I would argue that we should not return views in `torch.no_grad()` mode.
Because here we "autodiff" the view op even though it was in a no grad block.
See https://github.com/pytorch/pytorch/issues/26546 for reference of this problem.

Reply:
oooooh this is giving me flashbacks to another issue. This one: https://github.com/pytorch/pytorch/pull/12502 (and also #11390)